### ISSUE TITLE
Wrongs about CUDNN

### ISSUE THREAD

**Issue by heiyuxiaokai** [DetectionTeamUCAS/FPN_Tensorflow#4]
I met this problem when I tried to train for my dataset.
F tensorflow/stream_executor/cuda/cuda_dnn.cc:444] could not convert BatchDescriptor {count: 0 feature_map_count: 256 spatial: 14 14  value_min: 0.000000 value_max: 0.000000 layout: BatchDepthYX} to cudnn tensor descriptor: CUDNN_STATUS_BAD_PARAM
Aborted (core dumped)

**Comment by yangxue0827** [comment]
what is you cuda version @heiyuxiaokai

**Comment by heiyuxiaokai** [comment]
CUDA Version 8.0.61
CUDNN Version 6.0.21 
@yangxue0827

**Comment by yangxue0827** [comment]
if you use cuda8, you need uncommend [these code](https://github.com/DetectionTeamUCAS/FPN_Tensorflow/blob/master/libs/networks/build_whole_network.py#L225) @heiyuxiaokai

**Comment by heiyuxiaokai** [comment]
But my tensorflow version is 1.4.0. I tried to uncommend these code, and these is another problem.
It seems the GPU Memory run out.
When I change these para:
1.config = tf.Configproto() ——>config = tf.Configproto(allow_soft_placement = True)
2.config.gpu_options.allow_growth = True ——> False
train process can run several steps. 
(My GPU is TITAN x)
And when i use CPU to train, it runs orderly.
@yangxue0827

**Comment by heiyuxiaokai** [comment]
It works, Thanks! @yangxue0827

**Comment by yangxue0827** [comment]
Does the accuracy under cuda8 drop? @heiyuxiaokai

**Comment by heiyuxiaokai** [comment]
The loss dropped totality in a way. After step 120000, in some steps, the loss is about 0.5. But in other steps, the loss is high, which ranges from 1 to 5.
I tested the val dataset in step 120000. The accuracy is very low. Most classes can't be detected, and  the objects' confident score are low.
Have you tested the code under cuda8 and got a good accuracy?
@yangxue0827

**Comment by yangxue0827** [comment]
i will try. @heiyuxiaokai

**Comment by heiyuxiaokai** [comment]
Actually, I met the same problem when I tested the val under cuda8. #6
I guessed,  the reason might be:
I only changed the train code to fit cuda8 without changing the test code.
So I made some attempts:
https://github.com/DetectionTeamUCAS/FPN_Tensorflow/blob/07fd7b7ea9e91c94fce28a52998931a7d4c82945/libs/networks/build_whole_network.py#L243 to
`return tf.concat([level_i_rois, [[0, 0, 0., 0.]]], axis=0)`

The test.py could run well. But the result was not good.
@yangxue0827

**Comment by heiyuxiaokai** [comment]
There might be something wrong in test code under cuda8 instead of  wrong images.
@yangxue0827 @zxmfgky

**Comment by zxmfgky** [comment]
Got it, thanks! @heiyuxiaokai @yangxue0827

**Comment by yangxue0827** [comment]
已经改好了cuda版本的问题，在cuda8环境下的测试结果和cuda9基本一致。 @heiyuxiaokai @zxmfgky

**Comment by heiyuxiaokai** [comment]
我在修改多GPU训练时，那个正则化损失是应该计算一个gpu还是计算两个gpu
slim的源码注释说只计算一个gpu的正则化损失，但是我实验过后精度没有单gpu高，可能是迭代次数不够
改成两个GPU的正则化损失之后，完全识别不出来
想请作者给一点建议 @yangxue0827
